### DISCUSSION TITLE
wondering if there is a way to acquaint chatgpt to the workspace codebase?

### DISCUSSION THREAD

**Discussion by lauralee00** [acheong08/ChatGPT#81] [General]
I'm just coming across this API and there doesn't seem to be documentation on this so starting a discussion. Does every API call use independent sessions as opposed to a continued "thread" of questions that consider previous context we might see from a web based interaction with ChatGPT? Also, on the related note, I'm wondering if there is a way to systematically feed in the workspace code and structure to ChatGPT. Thanks!

**Comment by acheong08** [comment]
The thread is handled by the API's internals via conversation_id and parent_id so I don't need a single unbroken session. However, that is a point of improvement to increase code efficiency. I will refactor at some point but a bit busy atm

**Reply by dwimble** [reply]
you're doing a phenomenal job on this project well done!

**Comment by yxd0018** [comment]
@lauralee00 I'm searching the answer for the same question to feed my data set. Have you found a way? Thx

**Reply by acheong08** [reply]
By simply feeding it to the same Chatbot object, the thread is continued. To systematically feed data into ChatGPT, read https://github.com/acheong08/ChatGPT/wiki/Developer-Docs

**Reply by yxd0018** [reply]
Thanks a lot. Do you know if 
1. how I can provide feedback to make it a bit more supervised training?
2. is there a way to make newly trained bot remain private? The model seems not big.
